dsa-23au/java-dsa/sorting/src/main/java/dev/codewithfriends/App.java

```diff
@@ -34,9 +34,39 @@ public static int[] mergeSort(int[] array){
     public static int[] mergeSortHelper(int[] arr, int start, int end){
         int[] newArray = new int[end - start + 1]; 
         int length = end - start + 1;
+       
+        if (end - start > 1){
+            int leftEnd = start + length / 2;
+            int rightStart = leftEnd + 1;
+            mergeSortHelper(arr, start, leftEnd);       // left half
+            mergeSortHelper(arr, rightStart, end);     // right half
+            
+            for (int i = start; i < rightStart; i++){
+                newArray[i - start] = arr[i];
+            }
 
-        mergeSortHelper(arr, start, start + (length / 2));       // left half
```

Comment:
I think this was removed and changed on line 41 and 42 to make it more simple since  leftEnd = start + length / 2.